client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker-preview.tsx

```diff
@@ -607,22 +608,32 @@ const UnifiedDesignPickerPreview = ( {
 				/>
 			) }
 			<QueryEligibility siteId={ site?.ID } />
-			<EligibilityWarningsModal
-				site={ site ?? undefined }
-				isMarketplace={ selectedDesign?.is_externally_managed }
-				isOpen={ showEligibility }
-				handleClose={ () => {
-					recordTracksEvent( 'calypso_automated_transfer_eligibility_modal_dismiss', {
-						flow: 'onboarding',
-						theme: selectedDesign?.slug,
-					} );
-					setShowEligibility( false );
-				} }
-				handleContinue={ () => {
-					navigateToCheckout();
-					setShowEligibility( false );
-				} }
-			/>
+			{ showEligibility && (
+				<Modal
+					className="eligibility-warnings-modal__dialog-content"
+					title={ translate( 'Before you continue' ) }
+					onRequestClose={ () => {
+						recordTracksEvent( 'calypso_automated_transfer_eligibility_modal_dismiss', {
+							flow: 'onboarding',
+							theme: selectedDesign?.slug,
+						} );
+						setShowEligibility( false );
+					} }
```

Comment:
We'll need to extract this function and also pass it to the `EligibilityWarnings` component via the `onDismiss` prop, to close the modal when loading Odie.

Reply:
Done!